src/core/lib/surface/completion_queue.c

```diff
@@ -576,6 +549,42 @@ static void cq_check_tag(grpc_completion_queue *cq, void *tag, bool lock_cq) {
 static void cq_check_tag(grpc_completion_queue *cq, void *tag, bool lock_cq) {}
 #endif
 
+static bool cq_begin_op_for_next(grpc_completion_queue *cq, void *tag) {
+  cq_next_data *cqd = DATA_FROM_CQ(cq);
+  while (true) {
+    gpr_atm count = gpr_atm_no_barrier_load(&cqd->pending_events);
+    if (count == 0) {
+      cq_check_tag(cq, tag, true); /* Used in debug builds only */
```

Comment:
wouldn't `cq_check_tag` fail here since the cq is empty  (i.e count == 0 would mean cq is shutdown and there are no tags. Whereas cq_check_tag expects the tag to be present in cq and would assert otherwise)